radar-auth/src/main/java/org/radarcns/auth/authorization/RadarAuthorization.java

```diff
@@ -80,35 +69,25 @@ public static void checkPermissionOnSubject(DecodedJWT token, Permission permiss
             String projectName, String subjectName) {
         log.debug("Checking permission {} for user {} on subject {} in project {}",
                 permission.toString(), token.getSubject(), subjectName, projectName);
-        if (isSuperUser(token) || hasScope(token, permission)) {
-            return;
-        }
-        // we're allowed to read our own data
-        if (token.getSubject().equals(subjectName) && Permissions.allowedAuthorities(permission)
-                .contains(AuthoritiesConstants.PARTICIPANT)) {
-            return;
-        }
-        // if we're only a participant, and we're not the subject we request data for,
-        // we don't have access
-        if (isJustParticipant(token, projectName)) {
-            throw new NotAuthorizedException(String.format("User %s does not have permission %s in "
-                    + "project %s for subject %s", token.getSubject(), permission.toString(),
-                    projectName, subjectName));
-        } else {
-            // otherwise we have other roles and we should check on a project level
-            checkPermissionOnProject(token, permission, projectName);
-        }
-    }
+        checkScope(token, permission);
+        if (!isClientCredentials(token)) {
+            // we're allowed to read our own data
+            if (token.getSubject().equals(subjectName) && Permissions.allowedAuthorities(permission)
```

Comment:
Is this api used by our clients? Since `SubjectDTO` exposed quite some data about the sources, projects, external ids etc. I wonder, what would if by any chance a valid refresh-token is exposed, which i think could be possible then they will get all the above details. I would say subject shouldn't be able to read on the other hand i also think, `subject/{login/sources` is used for `ROLE_PARTICIPANT` for `pRMT`. Maybe we can think of a processed meta-data for the future.

Reply:
For future reference: after offline discussion we decided to keep the current permissions for `ROLE_PARTICIPANT`, as not to break the apps. We'll open a discussion with the wider group to get a consensus on the permissions a participant should and should not have at a later time.